### ISSUE TITLE
[Flaky test] [k8s.io] [sig-node] Kubelet [Serial] [Slow] [k8s.io] [sig-node] regular resource usage tracking resource tracking for 0 pods per node

### ISSUE THREAD

**Issue by mariantalla** [kubernetes/kubernetes#75298]
**Which jobs are failing**:
* `ci-kubernetes-e2e-gce-new-master-upgrade-master`
* `ci-kubernetes-e2e-gce-new-master-upgrade-cluster-new`

**Which test(s) are failing**:
`[k8s.io] [sig-node] Kubelet [Serial] [Slow] [k8s.io] [sig-node] regular resource usage tracking resource tracking for 0 pods per node`

**Testgrid link**:
https://testgrid.k8s.io/sig-release-master-upgrade#gce-new-master-upgrade-cluster-new&width=20

**Reason for failure**:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/node/kubelet_perf.go:263
Mar 11 11:51:31.967: Memory usage exceeding limits:
 node bootstrap-e2e-minion-group-rrbx:
 container "runtime": expected RSS memory (MB) < 131072000; got 158027776
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/node/kubelet_perf.go:155
```

**Anything else we need to know**:
* Impact:
  * Occasional failures make the entire release-blocking job fail
  * It seems that some of the time, kubelet's memory usage exceeds expected limits. Is that normal/indicate something wrong with that area of the code? If not, can the limits be adjucted to make the test more stable?
* this test flakes about 4% of the time in master-upgrade, up to 30% in other jobs
* [triage - across all jobs](https://storage.googleapis.com/k8s-gubernator/triage/index.html?test=regular%20resource%20usage%20tracking%20resource%20tracking%20for%200%20pods%20per%20node)
* [triage - for `ci-kubernetes-e2e-gce-new-master-upgrade-cluster-new`
](https://storage.googleapis.com/k8s-gubernator/triage/index.html?job=ci-kubernetes-e2e-gce-new-master-upgrade-cluster-new&test=regular%20resource%20usage%20tracking%20resource%20tracking%20for%200%20pods%20per%20node)

/kind flake
/remove-kind failing-test
/sig node
/priority important-soon

@dashpole could you have a look please?

**Comment by mariantalla** [comment]
Adding it to 1.14 for now, but it's lower priority than https://github.com/kubernetes/kubernetes/issues/75039.

/milestone v1.14

cc @smourapina @alejandrox1 @kacole2 @mortent

**Comment by mariantalla** [comment]
Hi @dashpole - could you have a look please? This just failed in `gce-cos-master-serial`: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-serial/4779

**Comment by dashpole** [comment]
This looks like the same as #75039.  There are pods running on the node that are not from this test.  That very likely is the reason the kubelet has higher memory usage in some runs:

```
Logging pods the kubelet thinks is on node bootstrap-e2e-minion-group-rrbx
ds1-cklv4
metadata-proxy-v0.1-h5h92
rs-vsc7t
coredns-5b969f4c88-pgvkp
echoheaders-https-bkzff
event-exporter-v0.2.3-fdd9d7c84-x8ntg
echoheaders-https-ml9bt
kubernetes-dashboard-66bb48f98c-5lrm8
ss-0
ss-2
kube-dns-autoscaler-97df449df-tlswn
foo-2fmzn
res-cons-upgrade-ctrl-pb2h4
fluentd-gcp-scaler-85cdd7d7c5-2t7qn
kube-proxy-bootstrap-e2e-minion-group-rrbx 
fluentd-gcp-v3.2.0-6dtwv
foo-gtrv7
service-test-klspp
metrics-server-v0.3.1-58d65f8d6-wlq94
dp-64d77b4887-pgwrx
```

**Comment by mariantalla** [comment]
Hey @dashpole - is there anything else we can do to figure out where the other pods are coming from? It's a bit weird as these jobs are serial (I think at least). 🤔 

fwiw it's also flaking about 15% of the time in `ci-kubernetes-e2e-gce-new-master-upgrade-cluster`

**Comment by dashpole** [comment]
Those pods are from other tests.  For example, the "ds1" pod is from the [daemonset upgrade test](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/upgrades/apps/daemonsets.go#L43).  Reading through the test, it looks like it is intentionally being run in parallel with the serial test.  I.E. The daemonset upgrade test creates a daemonset, runs a bunch of other tests, and then verifies the daemonset is still running.

If this is the case, we probably shouldn't run the kubelet benchmark test with upgrade tests in general.  The whole point of the test is to run it in a controlled environment with a specific number of pods.

**Comment by alejandrox1** [comment]
Still need to keep an eye on this one
/remove-milestone v1.14
/milestone v1.15
